Source/Classes/PINRemoteImageManager.m

```diff
@@ -784,13 +786,62 @@ - (void)downloadImageWithURL:(NSURL *)url
             NSError *remoteImageError = error;
             PINImage *image = nil;
             id alternativeRepresentation = nil;
-             
+            __block NSNumber *maxAge = nil;
             if (remoteImageError == nil) {
-                 //stores the object in the caches
-                 [self materializeAndCacheObject:data cacheInDisk:data additionalCost:0 url:url key:key options:options outImage:&image outAltRep:&alternativeRepresentation];
+                if (_isTtlCache && !(options & PINRemoteImageManagerDownloadOptionsIgnoreCacheControlHeaders)) {
+                    // examine Cache-Control headers (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+                    if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
+
+                        [[[httpResponse allHeaderFields][@"Cache-Control"] componentsSeparatedByString:@","] enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+                            NSString *trimmed = [[(NSString *) obj stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]] lowercaseString];
+
+                            if ([trimmed isEqualToString:@"no-store"] || [trimmed isEqualToString:@"must-revalidate"] || [trimmed isEqualToString:@"no-cache"]) {
+                                maxAge = @(0);
+                                *stop = YES;
+                            } else {
+                                // max-age
+                                NSArray<NSString *> *split = [trimmed componentsSeparatedByString:@"max-age="];
+                                if ([split count] == 2) {
+                                    maxAge = @([split[1] integerValue]);
+                                }
+                            }
+                        }];
+
+                        NSString *expires = nil;
+                        // If there is a Cache-Control header with the "max-age" directive in the response, the Expires header is ignored
+                        if (!maxAge && (expires = [httpResponse allHeaderFields][@"Expires"])) {
+                            // https://developer.apple.com/library/archive/qa/qa1480/_index.html
+
+                            static NSDateFormatter *sRFC7231PreferredDateFormatter;
```

Comment:
Assign nil. I'm not sure we should rely on static variables being zero-initialized.

Also, we should move this out to an accessor method. Ideally static variables are declared outside methods (even if just above the method). In this case a relatively complex method would be shortened if we moved the cached formatter to an accessor like +preferredCacheControlDateFormatter.